core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2Metadata.java

```diff
@@ -184,19 +186,20 @@ private String getToken() throws IOException {
         AbortableInputStream abortableInputStream = null;
         try {
             URI uri = URI.create(endpoint + TOKEN_RESOURCE_PATH);
-            HttpExecuteRequest httpExecuteRequest = requestMarshaller.createTokenRequest(uri , SdkHttpMethod.PUT, tokenTtl);
-            HttpExecuteResponse response =  httpClient.prepareRequest(httpExecuteRequest).call();
+            HttpExecuteRequest httpExecuteRequest = REQUEST_MARSHALLER.createTokenRequest(uri, SdkHttpMethod.PUT, tokenTtl);
+            HttpExecuteResponse response = httpClient.prepareRequest(httpExecuteRequest).call();
             int statusCode = response.httpResponse().statusCode();
 
-            if (statusCode == HttpURLConnection.HTTP_OK) {
+            if (statusCode == HttpURLConnection.HTTP_OK && response.responseBody().isPresent()) {
                 abortableInputStream = response.responseBody().get();
                 String token = IoUtils.toUtf8String(abortableInputStream);
                 return token;
             } else if (statusCode == HttpURLConnection.HTTP_FORBIDDEN || statusCode == HttpURLConnection.HTTP_BAD_REQUEST) {
-                throw SdkServiceException.builder().message("Could not retrieve token").build();
+                throw SdkServiceException.builder()
+                                         .message("Could not retrieve token as " + statusCode + " error occurred.").build();
             } else {
                 throw SdkClientException.builder()
-                                        .message("Unexpected Exception during token retrieval with Status Code " + statusCode)
+                                        .message("Instance metadata service returned unexpected status code " + statusCode)
```

Comment:
If the response body isn't present then we'd get unexpected status code 200. Should we special-case the response body not being present with a better error message?